server/sonar-server/src/main/java/org/sonar/ce/http/CeHttpClientImpl.java

```diff
@@ -0,0 +1,188 @@
+  }
+
+  private static final class ChangeLogLevelActionClient implements ActionClient<Void> {
+    private final LoggerLevel newLogLevel;
+
+    private ChangeLogLevelActionClient(LoggerLevel newLogLevel) {
+      this.newLogLevel = newLogLevel;
+    }
+
+    @Override
+    public String getPath() {
+      return PATH_CHANGE_LOG_LEVEL;
+    }
+
+    @Override
+    public Void getDefault() {
+      return null;
+    }
+
+    @Override
+    public Void call(String url) throws Exception {
+      okhttp3.Request request = new okhttp3.Request.Builder()
+        .post(RequestBody.create(null, new byte[0]))
+        .url(url + "?level=" + newLogLevel.name())
+        .build();
+      okhttp3.Response response = new OkHttpClient().newCall(request).execute();
+      if (response.code() != 200) {
+        throw new IOException(
+          String.format(
+            "Failed to change log level in Compute Engine. Code was '%s' and response was '%s' for url '%s'",
+            response.code(),
+            response.body().string(),
+            url));
+      }
+      return null;
+    }
+  }
+
+  @Override
+  public void refreshCeWorkerCount() {
+    call(RefreshCeWorkerCountActionClient.INSTANCE);
+  }
+
+  private enum RefreshCeWorkerCountActionClient implements ActionClient<Void> {
+    INSTANCE;
+
+    @Override
+    public String getPath() {
+      return "refreshWorkerCount";
+    }
+
+    @Override
+    public Void getDefault() {
+      return null;
+    }
+
+    @Override
+    public Void call(String url) throws Exception {
+      okhttp3.Request request = new okhttp3.Request.Builder()
+        .post(RequestBody.create(null, new byte[0]))
+        .url(url)
+        .build();
+      okhttp3.Response response = new OkHttpClient().newCall(request).execute();
+      if (response.code() != 200) {
+        throw new IOException(
+          String.format(
+            "Failed to trigger refresh of CE Worker count. Code was '%s' and response was '%s' for url '%s'",
+            response.code(),
+            response.body().string(),
+            url));
+      }
+      return null;
+    }
+  }
+
+  private <T> T call(ActionClient<T> actionClient) {
+    try (DefaultProcessCommands commands = DefaultProcessCommands.secondary(ipcSharedDir, COMPUTE_ENGINE.getIpcIndex())) {
+      if (commands.isUp()) {
+        return actionClient.call(commands.getHttpUrl() + "/" + actionClient.getPath());
+      }
+      return actionClient.getDefault();
+    } catch (Exception e) {
+      throw new IllegalStateException("Failed to call HTTP server of process " + COMPUTE_ENGINE, e);
+    }
+  }
+
+  private interface ActionClient<T> {
+    /**
+     * Path of the action.
+     */
+    String getPath();
+
+    /**
+     * Value to return when the Compute Engine is not ready.
+     */
+    T getDefault();
+
+    /**
+     * Delegates to perform the call to the Compute Engine's specified absolute URL.
+     */
+    T call(String url) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)